src/streamlink/plugins/eltrecetv.py

```diff
@@ -13,27 +12,31 @@ def can_handle_url(cls, url):
         return cls._url_re.match(url)
 
     def _get_streams(self):
-
-        video_url_found_hls = ""
-
         if "eltrecetv.com.ar/vivo" in self.url.lower():
-            http.headers = {'Referer': self.url,
-            'User-Agent': "ArtearPlayer/3.2.42 (Linux;Android 4.4.2) ExoPlayerLib/1.5.9"}
-            video_url_found_hls = "http://stream.eltrecetv.com.ar/live13/13tv/13tv1/playlist.m3u8"
+            try:
+                http.headers = {'Referer': self.url,
+                'User-Agent': "ArtearPlayer/3.2.42 (Linux;Android 4.4.2) ExoPlayerLib/1.5.9"}
+                res = http.get('https://api.iamat.com/metadata/atcodes/eltrece')
+                live_search = res.text
+                live_search = live_search[live_search.index('"vivo":{"provider":"youtube","youtubeVideo":')+45 :]
+                live_search = live_search[: live_search.index('"')]
+                LIVE_URL_FOUND = 'https://www.youtube.com/watch?v=' + live_search
+                return self.session.streams(LIVE_URL_FOUND)
+            except:
+                self.logger.info("Live content is temporarily unavailable. Please try again later.")
         else:
-            http.headers = {'Referer': self.url,
-            'User-Agent': useragents.CHROME}
-            res = http.get(self.url)
-            _player_re = re.compile(r'''data-kaltura="([^"]+)"''')
-            match = _player_re.search(res.text)
-            if not match:
-                return
-            json_video_search = parse_json(match.group(1).replace("&quot;", '"'))
-            video_url_found_hls = "https://vodgc.com/p/111/sp/11100/playManifest/entryId/" + json_video_search["entryId"] + "/format/applehttp/protocol/https/a.m3u8"
-
-        if video_url_found_hls:
-            hls_streams = HLSStream.parse_variant_playlist(self.session, video_url_found_hls)
-            for s in hls_streams.items():
-                yield s
+            try:
+                http.headers = {'Referer': self.url,
+                'User-Agent': useragents.CHROME}
+                res = http.get(self.url)
+                _player_re = re.compile(r'''data-kaltura="([^"]+)"''')
+                match = _player_re.search(res.text)
+                if not match:
+                    return
+                json_video_search = parse_json(match.group(1).replace("&quot;", '"'))
+                VOD_URL_FOUND = "https://vodgc.com/p/111/sp/11100/playManifest/entryId/" + json_video_search["entryId"] + "/format/applehttp/protocol/https/a.m3u8"
+                return self.session.streams(VOD_URL_FOUND)
```

Comment:
This should really just use `HLSStream` or `HLSStream.parse_variant_playlist`, which ever applies.